inspirehep/modules/editor/views.py

```diff
@@ -225,3 +228,27 @@ def _simplify_ticket_response(ticket):
         owner=ticket['Owner'],
         date=ticket['Created'],
         link=ticket['Link'])
+
+
+@blueprint.route('/upload', methods=['POST'])
+@editor_use_api_permission.require(http_exception=403)
+def upload_files():
+    # check if the post request has the file part
+    if 'file' not in request.files:
+        return jsonify(success=False, message='No file part')
+    attachment = request.files['file']
+    # if user does not select file, browser also
+    # submit a empty part without filename
+    if attachment:
+        filedir = current_app.config['RECORD_EDITOR_FILE_UPLOAD_FOLDER']
+        if not os.path.isdir(filedir):
+            os.makedirs(filedir)
+        filename = secure_filename(attachment.filename)
+        filepath = os.path.join(filedir, filename)
+        if os.path.isfile(filepath):
+            count = 1
```

Comment:
Since this is the second time this pattern is going to be used (first one https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/api.py#L488-L495) it would be good to add a utility function for it and that would make it also easier to test.

Reply:
Is it important to conserve the name passed from the UI or append an index to the name? Otherwise by using the filename as sufix in https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp you could avoid checking for duplicate files.

Reply:
This name I think would be in the final url for the file and we needed it to be user readable. Where should I put the utility function you think?

Reply:
we could create `utils/files.py` for filename/directory related utils

Reply:
@jmartinm I used this pattern in a bit different way and in my opinion the pattern inside the while loop is not worth moving to a util function. I tried to integrate the whole loop into a function but the conditions in the 2 cases are different.